src/devices/bus/isa/ibm_speech.cpp

```diff
@@ -0,0 +1,467 @@
+// license:BSD-3-Clause
+// copyright-holders:Katherine Rohl
+/*
+    IBM PS/2 Speech Adapter
+
+    The ROM listing calls this card the "TALKER II".
+
+    The ISA version of the card uses a different timing mechanism
+    than the PCjr version to account for a 4MHz operating speed
+    rather than the 4.77 bus clock. The ISA version also uses
+    IRQ7 rather than IRQ1.
+
+    Lots of TTL, all off-the-shelf parts except the mask ROM.
+
+    Y1      - 4MHz oscillator
+    ZM5     - MC14529CP
+    ZM8     - TMS5220CNL
+    ZM9     - 8254 PIT
+    ZM10    - 32Kx8 mask ROM
+    ZM22    - 8255A PIC
+*/
+
+#include "emu.h"
+#include "xtal.h"
+#include "ibm_speech.h"
+
+#define LOG_ROM         (1 << 0)
+#define LOG_MUX         (1 << 1)
+#define LOG_CVSD        (1 << 2)
+#define LOG_PIT         (1 << 3)
+#define LOG_IRQ         (1 << 4)
+#define LOG_PORTS       (1 << 5)
+#define LOG_ALL         (LOG_ROM|LOG_MUX|LOG_CVSD|LOG_PIT|LOG_IRQ|LOG_PORTS)
+
+//#define VERBOSE         (LOG_PORTS)
+#include "logmacro.h"
+
+#ifdef _MSC_VER
+#define FUNCNAME __func__
+#else
+#define FUNCNAME __PRETTY_FUNCTION__
+#endif
+
+#define XTAL_Y1     XTAL(4'000'000)
```

Comment:
Can you make this a `constexpr` local in the function where it’s used?  In general, preprocessor macros are more likely to cause trouble.